File: budgette-frontend/src/types/account.ts
export type Operator = 'MTN' | 'MOOV'
export type Country = 'BJ'

export interface Account {
  id: string
  operator: Operator
  operatorDisplayName: string
  country: Country
  phoneNumber: string
  balance: number
  currency: string
  active: boolean
}

export interface AddAccountRequest {
  operator: Operator
  country: Country
  phoneNumber: string
}

File: src/main/java/com/budgette/backend/domain/model/Account.java
package com.budgette.backend.domain.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Account {

    private String id;
    private String userId;
    private Operator operator;
    private Country country;
    private String phoneNumber;
    private String accountId;
    private BigDecimal balance;
    private boolean active;
}
